apis/cluster/v1/membercluster_types.go

```diff
@@ -0,0 +1,209 @@
+*/
+
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster,categories={fleet,fleet-cluster},shortName=cluster
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:JSONPath=`.status.conditions[?(@.type=="Joined")].status`,name="Joined",type=string
+// +kubebuilder:printcolumn:JSONPath=`.metadata.creationTimestamp`,name="Age",type=date
+
+// MemberCluster is a resource created in the hub cluster to represent a member cluster within a fleet.
+type MemberCluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// The desired state of MemberCluster.
+	// +required
+	Spec MemberClusterSpec `json:"spec"`
+
+	// The observed status of MemberCluster.
+	// +optional
+	Status MemberClusterStatus `json:"status,omitempty"`
+}
+
+// MemberClusterSpec defines the desired state of MemberCluster.
+type MemberClusterSpec struct {
+	// +kubebuilder:validation:Required,Enum=Join;Leave
+
+	// The identity used by the member cluster to access the hub cluster.
+	// The hub agents deployed on the hub cluster will automatically grant the minimal required permissions to this identity for the member agents deployed on the member cluster to access the hub cluster.
+	// +required
+	Identity rbacv1.Subject `json:"identity"`
+
+	// +kubebuilder:default=60
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=600
+
+	// How often (in seconds) for the member cluster to send a heartbeat to the hub cluster. Default: 60 seconds. Min: 1 second. Max: 10 minutes.
+	// +optional
+	HeartbeatPeriodSeconds int32 `json:"heartbeatPeriodSeconds,omitempty"`
+
+	// If specified, the MemberCluster's taints.
+	// +kubebuilder:validation:MaxItems=100
+	// +optional
+	Taints []Taint `json:"taints,omitempty"`
+}
+
+// PropertyName is the name of a cluster property; it should be a Kubernetes label name.
+type PropertyName string
+
+// PropertyValue is the value of a cluster property.
+type PropertyValue struct {
+	// Value is the value of the cluster property.
+	//
+	// Currently, it should be a valid Kubernetes quantity.
+	// For more information, see
+	// https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#Quantity.
+	//
+	// +required
+	Value string `json:"value"`
+
+	// ObservationTime is when the cluster property is observed.
+	// +required
+	ObservationTime metav1.Time `json:"observationTime"`
+}
+
+// MemberClusterStatus defines the observed status of MemberCluster.
+type MemberClusterStatus struct {
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+
+	// Conditions is an array of current observed conditions for the member cluster.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions"`
+
+	// Properties is an array of properties observed for the member cluster.
```

Comment:
This is a preview feature, I think we need to mark this filed as preview like k8s did

Reply:
Sure, I will add the comments. These would apply for v1beta1 as well I assume? (i.e., alpha level field)

Reply:
Also, do we do the same for other features, e.g., taints as well?

Reply:
Per earlier discussion, a beta-level hint comment is added.